### ISSUE TITLE
 Incorrect ligature box in NotoSansSiddham-Regular.otf

### ISSUE THREAD

**Issue by ijaradnam** [latex3/luaotfload#134]
Compiling with xelatex produces correct output, but lualatex-dev does not output boxes correctly.
`\documentclass{article}`
`\usepackage{fontspec}`
`\newfontfamily\sidd[Script=Siddham,Renderer=HarfBuzz]{Noto Sans Siddham Regular}`
`\begin{document}`
`\fboxsep=0pt`
`\fbox{\sidd 𑖨𑖿𑖔𑖿𑖎𑖿𑖎𑖿𑖎𑖿𑖎𑖻}`
`\end{document}`

**Comment by zauguin** [comment]
See https://mailman.ntg.nl/pipermail/dev-luatex/2020-March/006393.html for some context: Basically luatex *by design* does not change the depth of a box when glyph are moved down. 

An example using the ConTeXt fontloader and latin fonts would e.g. be

```
\documentclass{article}
\begin{document}
\font\pagella"TeX Gyre Pagella"\pagella
\fboxsep=0pt
\fbox{f^^^^0308^^^^0308^^^^0327^^^^0327}
\end{document}
```
![image](https://user-images.githubusercontent.com/962229/76632950-43544880-6544-11ea-9d80-1f71d56b5bdd.png)


Now I'm not sure what the best approach is for fixing this.

There will be a parameter available to change the handling in the engine starting with TeX Live 2021 (sic!), but I would prefer a faster fix. We could provide some callbacks which adjust depth calculations, but that wouldn't really fit into luaotfload, especially since they would have to operate after line breaking to be effective.

Maybe create a small package that can be loaded to change this? That could also be used as a test to decide if the parameter can be activated safely when it is available.

@u-fischer What do you think?

**Comment by u-fischer** [comment]
I think you are right and luaotfload is not the right place. A package sounds sensible. Perhaps we should start a "luacollection" package with such bits and pieces.

**Comment by zauguin** [comment]
@u-fischer Sounds good, I could also move `pre_append_to_vlist_filter.lua` and `docstrip-luacode.sty` there which currently live in `lua-ul`. I will create a repo under my account for now, then we can see later if it should be moved to LaTeX3.

**Comment by ijaradnam** [comment]
I have confirmed that this issue is resolved in luaTeX 1.13.
